app/src/main/java/org/surfsite/gexporter/MainActivity.java

```diff
@@ -671,20 +685,20 @@ protected void serveFiles() {
                             Log.error("Exception Open URI:", e);
                         }
                     }
+                } else {
+                    // try to open as file. Don't use DocumentFile.isFile() which seems to be unreliable when serving a gpx from a different app
+                    Log.debug("Open URI {} scheme {}", uri, scheme);
+                    try {
+                        InputStream is = mCR.openInputStream(uri);
+                        String name = getFileName(uri);
+                        copyInputStreamToFile(is, new File(mDirectory, name));
+                    } catch (FileNotFoundException e) {
+                        Log.error("Exception Open URI:", e);
+                    }
                 }
             }
         }
 
-        File directory;
-
-        if (rootDirectory.equals(downloadDirectory)) {
-            directory = Environment.getExternalStoragePublicDirectory(DIRECTORY_DOWNLOADS);
-            Log.error("Using DIRECTORY_DOWNLOADS");
-        } else {
-            directory = new File(rootDirectory);
-            Log.error("OpenDir {}", rootDirectory);
-        }
-
```

Comment:
seems like I forgot to push https://github.com/gimportexportdevs/gexporter/commit/dd5ebac2335e268814ed69fdd298fbcc5ee1988b after doing the V15 release... sigh